client/state/comments/reducer.js

```diff
@@ -191,6 +192,38 @@ export const totalCommentsCount = createReducer(
 );
 
 /**
+ * Stores a site's comments tree.
+ * @param {Object} state Redux state (`comments.trees`)
+ * @param {Object} action Redux action
+ * @returns {Object} New Redux state.
+ */
+export const trees = ( state = {}, { siteId, status, tree, type } ) => COMMENTS_TREE_RECEIVE === type
+	? {
+		...state,
+		[ siteId ]: {
+			...state[ siteId ],
+			[ status ]: tree,
```

Comment:
let's try and think about making _idempotent_ reducers that don't simply wipe out the history of what came before the action. think of a `merge()` operation which can funnel a tree in _associatively_ by parts.